add notes about what the job states mean (#130)
buildkite/buildkite-agent-metrics: README.md

When a queue is specified, only that queue's metrics are published.

> Jobs can be in one of the following states:

Reviewer: ```suggestion
We send metrics for Jobs in the following states:
```

We spoke a little bit publicly about the states a job can be in [here](https://www.buildkitestatus.com/incidents/k9nh8yy0px4q), this isn't a complete list of states, just the states we report metrics for.

Reviewer: Actually here's the public documentation for [Job States](https://buildkite.com/docs/pipelines/defining-steps#job-states)

* **Scheduled**, the job hasn't been assigned to an agent yet. If you have agent capacity, this value should be close to 0.

> * **Waiting**, the job is known to exist but isn't schedulable yet due to dependencies, `wait` statements, etc. This information is mostly useful to an autoscaler, since it represents work that will start soon.

Reviewer: This matches the PR description for when it was added #81
